.tools/create_algo_selection_code.py

```diff
@@ -119,47 +119,47 @@ def _get_algorithms_in_module(module: ModuleType) -> dict[str, Type[Algorithm]]:
 # Functions to filter algorithms by selectors
 # ======================================================================================
 def _is_gradient_based(algo: Type[Algorithm]) -> bool:
-    return algo.__algo_info__.needs_jac  # type: ignore
+    return algo.algo_info.needs_jac  # type: ignore
```

Comment:
So mypy still does not understand this. Any ideas why?

Reply:
I was surprised too. For mypy this is a method, even though it is a property.